sklearn/manifold/tests/test_spectral_embedding.py

```diff
@@ -507,4 +507,12 @@ def test_spectral_eigen_tol_auto(monkeypatch, solver):
 
     spectral_embedding(S, random_state=42, eigen_solver=solver, eigen_tol="auto")
     mocked_solver.assert_called()
+
+    print(mocked_solver.call_args)
+
+    args, kwargs = mocked_solver.call_args
+
+    print(kwargs)
+    print(type(kwargs))
+
     assert mocked_solver.call_args.kwargs["tol"] == default_value
```

Comment:
I suppose that the following should work in Python 3.7 (I cannot install on the M1 with conda-forge).

```suggestion

    _, kwargs = mocked_solver.call_args
    assert kwargs["tol"] == default_value
```

Reply:
yes that's my guess also but it does not correspond to the current error which seems to occur when trying to index by "tol"

Reply:
It depends on what `mocked_solver.call_args.kwargs` would return in Python 3.7. It might be unintended and provide a tuple and we try to index it with a string.

Reply:
![image](https://user-images.githubusercontent.com/7454015/171607244-bb410de9-4ff3-469d-a788-06cb499e85af.png)

Reply:
So not sure which API `_Call` is following but it is raising the error.
The change is working FYI.